test/Constraints/rdar38225184.swift

```diff
@@ -0,0 +1,4 @@
+// RUN: %target-typecheck-verify-swift
+
+extension Collection where Element == Int && Index == Int {}
+// expected-error {{use ',' for spliting conditions rather than '&&'}}
```

Comment:
Maybe add a test for the fix-it too. e.g. `{{43-45=,}}`

Reply:
Thanks a lot, I didn't mention the fix-it.